src/pokemon/dexService.ts

```diff
@@ -0,0 +1,40 @@
+import fs = require("fs");
+import FuzzyMatching = require('fuzzy-matching');
+import { toID } from "../app";
+import { Pokemon, Dex } from "./models";
+
+export class DexService {
+
+	private pokemonMap: { [name: string]: Pokemon } = {};
+	private dex: Dex = {} as Dex;
+	private fuzzyMatching: FuzzyMatching;
+
+	constructor() {
+		this.loadFileData();
+		this.fuzzyMatching = new FuzzyMatching(this.dex.pokemon.map(p => p.name));
+	}
+
+	public getPokemon(name: string): Pokemon | undefined {
+		const pokemon = this.pokemonMap[toID(name)];
+		if (pokemon)
+			return pokemon;
```

Comment:
Use braces, I plan to setup eslint to require that.

Reply:
Sure. List for that is a very good idea.